### ISSUE TITLE
Use CSW geocat to feed content of infobox

### ISSUE THREAD

**Issue by davidoesch** [geoadmin/mf-geoadmin3#3023]
prerequisite
- [ ] geocat team has a QS process in place to manually publish edits of owners in geocat

Then
- [ ]   title
- [ ]   short title
- [ ]   Abstract
- [ ]   id geo ig 
- [ ]   link to geocat
- [ ]   link to scale number
- [ ]   link to detailbeschreibung
- [ ]   link to fachportal
- [ ]   link to wms service
- [ ]   Datenstand
- [ ]   optional: url to legend

can be updated by geocat directly

**Comment by mariokeusen** [comment]
+100 for that in general
Except for the datenstand: what about the auto-updates?

**Comment by loicgasser** [comment]
@ltclm This already the case no? Geocat info is directly imported the BOD. Now, I don't if it works with the new version of GeoCat.

**Comment by ltclm** [comment]
@loicgasser That's already the case now, yes, the import scripts are working with the new geocat version.
@davidoesch what exactly do you want to change with this ticket?
Do you want the layer info to be fetched directly from the csw service?

**Comment by davidoesch** [comment]
yes "Do you want the layer info to be fetched directly from the csw service?"
-> but before we need the QS to be done on geocat side ( up to now done in BOD)

**Comment by ltclm** [comment]
we have to import / cache geocat in the bod, layer search sphinx indices have to be updated too with this information.
but yes the qs has to be done in geocat.

**Comment by loicgasser** [comment]
This would be only for the infobox then. I am afraid the service will be a little slower.
Also layer information is used to search for layers.
What is the benefit of such an action? BOD is deployed and updated after each deploy.

**Comment by gjn** [comment]
Can we test performance to get this info from GeoCat directly? What's the rest services address?

**Comment by ltclm** [comment]
The information can be extracted from the following service url:
p.e. 
**ch.are.alpenkonvention**
SERVICE URL: 
http://www.geocat.ch/geonetwork/srv/eng/csw?request=GetRecordById&elementSetName=full&outputFormat=application/xml&outputSchema=http://www.geocat.ch/2008/che&service=CSW&version=2.0.2&id=8698bf0b-fceb-4f0f-989b-111e7c4af0a4

the geocat2bod interface is using xpath to do that.

My concerns:
* If we use this service as data source for layer info, we will have to find a solution to update the sphinx layer search indices too.
* What happens if geocat is down?

**Comment by loicgasser** [comment]
From my point of view it's too much work for a less performant/stable solution.
With a deploy every week this year metadata will be updated often enough.

If we still decide to go with it, I see 2 acceptable solutions:
1. GeoCat service + front end templating
2. Direct connection to GeoCat DB + templating backend

It would be great to have a common DB for GeoCat and MapGeoAdmin.

**Comment by davidoesch** [comment]
-csw uptime is an issue, I agree we don't control the service itis a SAAS contract with c2c
- the issue is, that geocat2bod is not done every week by IGKS (I agree, we try to fix here a process issue with a technological "workaround"

**Comment by gjn** [comment]
Clear process issue here as it shouldn't be automated because QS needs to be assured.

When IGKS does the geocat2bod, is everything on prod updated?

**Comment by davidoesch** [comment]
With the next deploy - yes

**Comment by gjn** [comment]
Not to forget: we have the meta info from geocat also in the getCaps of our WMS and WMTS. So at least, we need to cache this information somewhere for performance reaseon.

But I could well imagine an auto-import each x hours into bod prod...

**Comment by ltclm** [comment]
?? i think this it is not an option to auto import and publish data content to map.geo.admin.ch from geocat without QS.

**Comment by gjn** [comment]
QS should be in geocat. That's why I said we have a process issue.

Currently, we have info in geocat and in map.geo that is not the same - that's really wrong on many levels.

**Comment by davidoesch** [comment]
We have geocat2bod which is a human driven process for QS. it is all in place

**Comment by gjn** [comment]
So what exactly is the change that this issue should address?

**Comment by davidoesch** [comment]
To introduce a qs process in geocat and therefore we can omit geocat2bod. Status: geocat3 : a qs has still  not been introduced. So this ticket is on hold.

**Comment by gjn** [comment]
"it is all in place" confused me :=). So we wait on geocat QS process....

**Comment by davidoesch** [comment]
View continue to wait .... Ping sca once in a while

**Comment by davidoesch** [comment]
I waited now for geocat... long

Still propose to update SPHINX and BOD at least once an hour: so changes done by data owner are then published faster.

**Comment by gjn** [comment]
So on their side, they are OK with it? QS is in place?

@ltclm @loicgasser This would, IMO, simply mean to automate geocat2bod. What do you think the effort for this is?

**Comment by gjn** [comment]
(btw: it would be great to be able to work on change sets instead of full update on each iteration. Not sure of geocat supports that though)

**Comment by davidoesch** [comment]
on their site: no QS is in place ---- but if we offer hourly updates a faulty entry can be  changed fast.

**Comment by gjn** [comment]
So we need to inform them about it before we do it. And everybody involved has to give the go.

Then we do it.

**Comment by ltclm** [comment]
what exactly do you want to change, discussion here is a bit confusing [1]
1. geocat ->bod_master
1. geocat ->bod_master->dev->int->prod (sphinxsearch and api3 services, layerinfo, etc.)
1.  geocat-> api3 services (without bod)
1. geocat -> wms-bgdi translation

geocat import is not only used for mf-geodmin, wms translation is also using this data.
hourly deploy of bod, sorry but i think nobody wants that.
can we move this to re4?

[1]
> yes "Do you want the layer info to be fetched directly from the csw service?"
-> but before we need the QS to be done on geocat side ( up to now done in BOD)

> Still propose to update SPHINX and BOD at least once an hour: so changes done by data owner are then published faster.

**Comment by gjn** [comment]
+1 for RE4.

If we do it, then correctly. This means that all services that is using this meta information are updated. This includes WMTS/WMS Get capabilities. It could also mean that services using this data are set up differently, or we extract it from code.

We have 2 important goals for RE4:
1) 1 source of truth
2) if source of truth changes content, existing 'caches' need to update automatically

That's wild, I know.

**Comment by boecklic** [comment]
@boecklic commented: +100 for that in general
Except for the datenstand: what about the auto-updates?

**Comment by boecklic** [comment]
@boecklic commented: @ltclm This already the case no? Geocat info is directly imported the BOD. Now, I don't if it works with the new version of GeoCat.

**Comment by boecklic** [comment]
@boecklic commented: @loicgasser That's already the case now, yes, the import scripts are working with the new geocat version.
@davidoesch what exactly do you want to change with this ticket?
Do you want the layer info to be fetched directly from the csw service?

**Comment by boecklic** [comment]
@boecklic commented: yes "Do you want the layer info to be fetched directly from the csw service?"
-> but before we need the QS to be done on geocat side ( up to now done in BOD)

**Comment by boecklic** [comment]
@boecklic commented: we have to import / cache geocat in the bod, layer search sphinx indices have to be updated too with this information.
but yes the qs has to be done in geocat.

**Comment by boecklic** [comment]
@boecklic commented: This would be only for the infobox then. I am afraid the service will be a little slower.
Also layer information is used to search for layers.
What is the benefit of such an action? BOD is deployed and updated after each deploy.

**Comment by boecklic** [comment]
@boecklic commented: Can we test performance to get this info from GeoCat directly? What's the rest services address?

**Comment by boecklic** [comment]
@boecklic commented: The information can be extracted from the following service url:
p.e. 
**ch.are.alpenkonvention**
SERVICE URL: 
http://www.geocat.ch/geonetwork/srv/eng/csw?request=GetRecordById&elementSetName=full&outputFormat=application/xml&outputSchema=http://www.geocat.ch/2008/che&service=CSW&version=2.0.2&id=8698bf0b-fceb-4f0f-989b-111e7c4af0a4

the geocat2bod interface is using xpath to do that.

My concerns:
- If we use this service as data source for layer info, we will have to find a solution to update the sphinx layer search indices too.
- What happens if geocat is down?

**Comment by boecklic** [comment]
@boecklic commented: From my point of view it's too much work for a less performant/stable solution.
With a deploy every week this year metadata will be updated often enough.

If we still decide to go with it, I see 2 acceptable solutions:
1. GeoCat service + front end templating
2. Direct connection to GeoCat DB + templating backend

It would be great to have a common DB for GeoCat and MapGeoAdmin.

**Comment by boecklic** [comment]
@boecklic commented: -csw uptime is an issue, I agree we don't control the service itis a SAAS contract with c2c
- the issue is, that geocat2bod is not done every week by IGKS (I agree, we try to fix here a process issue with a technological "workaround"

**Comment by boecklic** [comment]
@boecklic commented: Clear process issue here as it shouldn't be automated because QS needs to be assured.

When IGKS does the geocat2bod, is everything on prod updated?

**Comment by boecklic** [comment]
@boecklic commented: With the next deploy - yes

**Comment by boecklic** [comment]
@boecklic commented: Not to forget: we have the meta info from geocat also in the getCaps of our WMS and WMTS. So at least, we need to cache this information somewhere for performance reaseon.

But I could well imagine an auto-import each x hours into bod prod...

**Comment by boecklic** [comment]
@boecklic commented: ?? i think this it is not an option to auto import and publish data content to map.geo.admin.ch from geocat without QS.

**Comment by boecklic** [comment]
@boecklic commented: QS should be in geocat. That's why I said we have a process issue.

Currently, we have info in geocat and in map.geo that is not the same - that's really wrong on many levels.

**Comment by boecklic** [comment]
@boecklic commented: We have geocat2bod which is a human driven process for QS. it is all in place

**Comment by boecklic** [comment]
@boecklic commented: So what exactly is the change that this issue should address?

**Comment by boecklic** [comment]
@boecklic commented: To introduce a qs process in geocat and therefore we can omit geocat2bod. Status: geocat3 : a qs has still  not been introduced. So this ticket is on hold.

**Comment by boecklic** [comment]
@boecklic commented: "it is all in place" confused me :=). So we wait on geocat QS process....

**Comment by boecklic** [comment]
@boecklic commented: View continue to wait .... Ping sca once in a while

**Comment by boecklic** [comment]
@boecklic commented: I waited now for geocat... long

Still propose to update SPHINX and BOD at least once an hour: so changes done by data owner are then published faster.

**Comment by boecklic** [comment]
@boecklic commented: So on their side, they are OK with it? QS is in place?

@ltclm @loicgasser This would, IMO, simply mean to automate geocat2bod. What do you think the effort for this is?

**Comment by boecklic** [comment]
@boecklic commented: (btw: it would be great to be able to work on change sets instead of full update on each iteration. Not sure of geocat supports that though)

**Comment by boecklic** [comment]
@boecklic commented: on their site: no QS is in place ---- but if we offer hourly updates a faulty entry can be  changed fast.

**Comment by boecklic** [comment]
@boecklic commented: So we need to inform them about it before we do it. And everybody involved has to give the go.

Then we do it.

**Comment by boecklic** [comment]
@boecklic commented: what exactly do you want to change, discussion here is a bit confusing [1]
1. geocat ->bod_master
1. geocat ->bod_master->dev->int->prod (sphinxsearch and api3 services, layerinfo, etc.)
1.  geocat-> api3 services (without bod)
1. geocat -> wms-bgdi translation

geocat import is not only used for mf-geodmin, wms translation is also using this data.
hourly deploy of bod, sorry but i think nobody wants that.
can we move this to re4?

[1]
> yes "Do you want the layer info to be fetched directly from the csw service?"
-> but before we need the QS to be done on geocat side ( up to now done in BOD)

> Still propose to update SPHINX and BOD at least once an hour: so changes done by data owner are then published faster.

**Comment by boecklic** [comment]
@boecklic commented: +1 for RE4.

If we do it, then correctly. This means that all services that is using this meta information are updated. This includes WMTS/WMS Get capabilities. It could also mean that services using this data are set up differently, or we extract it from code.

We have 2 important goals for RE4:
1) one source of truth
2) if source of truth changes content, existing 'caches' need to update automatically

That's wild, I know.

**Comment by ltclm** [comment]
this is already implemented / realized 80%::
https://docs.google.com/spreadsheets/d/1SER0o9E-tBO6hhxpN4-PRAkR9TMbl01gowlJsc4k260/edit#gid=0

the services in red are not possible at the moment, we have to get rid of po files in chsi and hit the bod everytime a tooltip click is done, which is not a good idea. We need to cache somehow the db requests.

There was already an LBM EPIC exercise and discussion about that
https://jira.swisstopo.ch/browse/KOGIS_PRJ-770

**Comment by boecklic** [comment]
@boecklic commented: this is already implemented / realized 80%::
https://docs.google.com/spreadsheets/d/1SER0o9E-tBO6hhxpN4-PRAkR9TMbl01gowlJsc4k260/edit#gid=0

the services in red are not possible at the moment, we have to get rid of po files in chsi and hit the bod everytime a tooltip click is done, which is not a good idea. We need to cache somehow the db requests.

There was already an LBM EPIC exercise and discussion about that
https://jira.swisstopo.ch/browse/KOGIS_PRJ-770

**Comment by ltclm** [comment]
??

**Comment by boecklic** [comment]
@boecklic commented: ??
